### PR TITLE
SMA-242: Solidity 0.8.20

### PR DESCRIPTION
**Pull request by ankurdubey521**

### CODE REVIEW

**Comment by ankurdubey521** [comment]
Yeah sure, I've marked this as Draft.
I've made some adjustments to make everything compile, including keeping some test contracts as 0.8.17 and deploying them via the deployContract cheat instead of constructor

File: DerivedPriceFeedFlat.sol
Comment (Aboudjem) [inline DerivedPriceFeedFlat.sol:1]: can you split this into two files?

File: DerivedPriceFeedFlat.sol [hunk 1]
```diff
@@ -6,40 +6,41 @@
 pragma solidity ^0.8.0;
```
Comment (Aboudjem) [inline DerivedPriceFeedFlat.sol:6]: shouldn't be ^0.8.20?

File: contracts/token/adapters/UniswapExample.sol [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.17;
+pragma solidity ^0.8.20;
 pragma abicoder v2;
```
Comment (Aboudjem) [inline contracts/token/adapters/UniswapExample.sol:3]: I think this has been by activated by default since 0.8.0

**Review by Aboudjem — CHANGES_REQUESTED** [review]

**Comment by Aboudjem** [comment]
Could you please merge it into the 'develop' branch?

File: DerivedPriceFeedFlat.sol (hunk 1 above)
Comment (ankurdubey521) [reply DerivedPriceFeedFlat.sol]: This file shouldn't exists in the first place, deleted it

File: contracts/token/adapters/UniswapExample.sol (hunk 2 above)
Comment (ankurdubey521) [reply contracts/token/adapters/UniswapExample.sol]: Yup, removed it

**Review by Aboudjem — APPROVED** [review]
🚀

**Pull request merged**